### PR TITLE
Carets - Maria McGrew - Api Muncher

### PR DESCRIPTION
**Pull request by ghost**
# API Muncher
Congratulations! You're submitting your assignment!

## Comprehension Questions
| Question | Answer    |
| :------------- | :------------- |
| How did you go about exploring the Edamam API, how did you try querying the API? | I started by exploring the API documentation on the Edamam site. I then used Postman to enter different GET requests in order to determine what my API wrapper would need in order to get the desired recipe information. |
| Describe your API Wrapper.  How did you decide on the methods you created? | I created a method that searched and listed the search results and a method that gets the individual recipe and lists that recipe's details. |
| Describe an edge case or failure case test you wrote for your API Wrapper. | I wrote a test that sends an error if the recipe request is invalid.  |
| Explain how VCR aids in testing an API. | It makes a call to the API and then stores the results from that request in a cassette. That cassette is used when doing testing in order to limit the number of API calls. |
| What is the Heroku URL of your deployed application? | https://maria-api-muncher.herokuapp.com/  |

### CODE REVIEW

File: lib/edamam_api_wrapper.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        return []
+      end
+  end
+
+  def self.show_recipe(uri)
+    url = BASE_URL + "?app_id=" + ID + "&app_key=" + KEYS + "&r=" + RECIPE_URI_PREFIX + URI.encode(uri)
+
+    response = HTTParty.get(url)
+      if response
+        image = URI.encode(response[0]["image"])
```
Comment (CheezItMan) [inline lib/edamam_api_wrapper.rb]: you also need to do:

```ruby
if response[0]
```

File: test/controllers/recipes_controller_test.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  describe "show" do
+    it "returns the requested recipe" do
+      VCR.use_cassette("show_action") do
+        get recipe_path("recipe_7bf4a371c6884d809682a72808da7dc2")
+        must_respond_with :success
+      end
+    end
+
+    it "returns an error if it has a bad request" do
```
Comment (CheezItMan) [inline test/controllers/recipes_controller_test.rb:39]: This test throws an error because of the thing I noted in the wrapper.

File: app/controllers/recipes_controller.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+class RecipesController < ApplicationController
+  def welcome
+  end
+
+  def index
+    @recipes = EdamamApiWrapper.list_recipes
```
Comment (CheezItMan) [inline app/controllers/recipes_controller.rb]: This should take an argument from params!

**Comment by CheezItMan** [comment]
# API Muncher
## What We're Looking For

| Feature | Feedback    |
| :------------- | :------------- |
| **Core Requirements** |   |
| Git hygiene | Good commit messages, but more granular commits would be better. |
| Comprehension questions	| Check |
| **General** |  |
| Rails fundamentals (RESTful routing, use of named paths) | Check |
| Semantic HTML | Semantic HTML, but very simple styling |
| Errors are reported to the user | No flash notices |
| API Wrapper to handle the API requests | Check |
| Controller testing | Testing is there, but there are some issues, see my notes in the code. |
| Lib testing | Yes, with some small notes |
| Search Functionality | Working now |
| List Functionality | It works, but not styled |
| Show individual item functionality (link to original recipe opens in new tab) | Unstyled, but it works |
| **Styling** |  |
| Foundation Styling for responsive layout | Nothing |
| List View shows 10 items at a time/pagination | Working now! |
| The app is styled to create an attractive user interface | Nope |
| **API Features** | |
| The App attributes Edaman | Attribution is there, but not linked to Edaman  |
| The VCR casettes do not contain the API key | Check |
| **External Resources** |  |
| Link to deployed app on Heroku | Deployed, but not working. |
|  **Overall** | I understand you were sick and had a hard time getting this in.  I appreciate the effort. |
